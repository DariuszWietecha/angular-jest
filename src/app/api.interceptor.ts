import { Injectable } from '@angular/core';
import { HttpEvent, HttpInterceptor, HttpHandler, HttpRequest } from '@angular/common/http';
import { Observable } from 'rxjs/index';

@Injectable()
export class ApiInterceptor implements HttpInterceptor {
  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
    const baseUrl = document.getElementsByTagName('apibase')[0].id;
    const apiReq = req.clone({ url: `${baseUrl}${req.url}` });
    return next.handle(apiReq);
  }
}
