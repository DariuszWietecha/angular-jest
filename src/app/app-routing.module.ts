import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { HomeComponent } from './components/home/home.component';
import { CategoriesListComponent } from './components/categories/categories-list/categories-list.component';
import { CategoriesCreateEditComponent } from './components/categories/categories-create-edit/categories-create-edit.component';
import { CompaniesListComponent } from './components/companies/companies-list/companies-list.component';
import { CompaniesCreateEditComponent } from './components/companies/companies-create-edit/companies-create-edit.component';


const routes: Routes = [
  { path: '', redirectTo: '/home', pathMatch: 'full' },
  { path: 'home', component: HomeComponent },
  { path: 'views/:categoryId', component: HomeComponent },
  { path: 'categories', component: CategoriesListComponent },
  { path: 'categories/:type', component: CategoriesCreateEditComponent },
  { path: 'categories/:type/:id', component: CategoriesCreateEditComponent },
  { path: 'companies', component: CompaniesListComponent },
  { path: 'companies/:type', component: CompaniesCreateEditComponent },
  { path: 'companies/:type/:id', component: CompaniesCreateEditComponent },
  { path: '**', component: HomeComponent },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
