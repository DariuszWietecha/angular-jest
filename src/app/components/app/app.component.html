<div class="container">
  <nav class="navbar navbar-expand-lg navbar-light bg-light">
    <a routerLink="/home" class="display-4">Angular-crud</a>
    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent"
      aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav mr-auto">
        <li class="nav-item">
          <a [routerLink]="['/home']" class="nav-link">Home</a>
        </li>
        <li class="nav-item">
          <a [routerLink]="['/companies']" class="nav-link">Companies</a>
        </li>
        <li class="nav-item">
          <a [routerLink]="['/categories']" class="nav-link">Categories</a>
        </li>
      </ul>
    </div>
  </nav>
  <div class="jumbotron">
    <router-outlet></router-outlet>
  </div>
</div>