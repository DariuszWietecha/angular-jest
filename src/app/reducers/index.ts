import {
  ActionReducer,
  ActionReducerMap,
  createFeatureSelector,
  createReducer,
  createSelector,
  MetaReducer,
  on
} from '@ngrx/store';
import { environment } from '../../environments/environment';
import { ICompany } from '../services/companies.service';
import { ICategory } from '../services/categories.service';
import * as categoriesActions from '../actions/categories.actions';
import * as companiesActions from '../actions/companies.actions';
import * as viewsActions from '../actions/views.actions';

export interface IStore {
  categories: ICategory[];
  companies: ICompany[];
  categoryId: string;
  editedCategory: ICategory | {};
  editedCompany: ICompany | {};
}

const categoriesReducer = createReducer([],
  on(viewsActions.loadHomeDataSuccess, (state, payload) => payload.categories),
  on(categoriesActions.loadCategoriesSuccess, (state, payload) => payload.data),
);

const companiesReducer = createReducer([],
  on(viewsActions.loadHomeDataSuccess, (state, payload) => payload.companies),
  on(companiesActions.loadCompaniesSuccess, (state, payload) => payload.data),
);

const categoryIdReducer = createReducer('',
  on(viewsActions.loadHomeDataSuccess, (state, payload) => payload.categoryId),
);

const editedCategoryReducer = createReducer({} as ICategory,
  on(categoriesActions.loadCategoriesSuccess, () => ({})),
  on(categoriesActions.loadCategorySuccess, (state, payload) => payload),
);

const editedCompanyReducer = createReducer({} as ICompany,
  on(companiesActions.loadCompaniesSuccess, () => ({})),
  on(companiesActions.loadCompanySuccess, (state, payload) => payload),
);

export const reducers: ActionReducerMap<IStore> = {
  categories: categoriesReducer,
  companies: companiesReducer,
  categoryId: categoryIdReducer,
  editedCategory: editedCategoryReducer,
  editedCompany: editedCompanyReducer,
};

export const selectHomeData = (state: IStore) => state;
export const selectCompaniesData = (state: IStore) => state.companies;
export const selectCategoriesData = (state: IStore) => state.categories;
export const selectEditedCategory = (state: IStore) => state.editedCategory;
export const selectEditedCompany = (state: IStore) => state.editedCompany;

export const metaReducers: MetaReducer<IStore>[] = !environment.production ? [] : [];
