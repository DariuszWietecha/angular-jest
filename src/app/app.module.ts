import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { AppRoutingModule } from './app-routing.module';
import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
import { ApiInterceptor } from './api.interceptor';
import { CompaniesService } from './services/companies.service';
import { CategoriesService } from './services/categories.service';
import { ViewsService } from './services/views.service';

import { AppComponent } from './components/app/app.component';
import { CategoriesListComponent } from './components/categories/categories-list/categories-list.component';
import { CategoriesCreateEditComponent } from './components/categories/categories-create-edit/categories-create-edit.component';
import { CompaniesListComponent } from './components/companies/companies-list/companies-list.component';
import { CompaniesCreateEditComponent } from './components/companies/companies-create-edit/companies-create-edit.component';
import { HomeComponent } from './components/home/home.component';
import { CompanyComponent } from './components/companies/company/company.component';

@NgModule({
  declarations: [
    AppComponent,
    HomeComponent,
    CategoriesListComponent,
    CategoriesCreateEditComponent,
    CompaniesListComponent,
    CompaniesCreateEditComponent,
    CompanyComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    HttpClientModule,
    ReactiveFormsModule,
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: ApiInterceptor, multi: true },
    CategoriesService,
    CompaniesService,
    ViewsService
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
