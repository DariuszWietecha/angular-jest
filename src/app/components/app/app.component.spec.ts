import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  }));

  beforeEach(async(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.debugElement.componentInstance;
    fixture.detectChanges();
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it(`should have as title 'angular-crud'`, () => {
    expect(component.title).toEqual('angular-crud');
  });

  it('should render title in display-4 class', () => {
    const compiledComponent = fixture.debugElement.nativeElement;
    expect(compiledComponent.querySelector('.display-4').textContent).toContain('Angular-crud');
  });

  it('snapShot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
