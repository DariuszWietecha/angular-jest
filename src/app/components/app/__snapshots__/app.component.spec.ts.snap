// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent snapShot 1`] = `
<app-root
  title={[Function String]}
>
  <div
    class="container"
  >
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light"
    >
      <a
        class="display-4"
        href="/home"
        ng-reflect-router-link="/home"
        routerlink="/home"
      >
        Angular-crud
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-target="#navbarSupportedContent"
        data-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          class="navbar-nav mr-auto"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/home"
              ng-reflect-router-link="/home"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/companies"
              ng-reflect-router-link="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/categories"
              ng-reflect-router-link="/categories"
            >
              Categories
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="jumbotron"
    >
      <router-outlet />
    </div>
  </div>
</app-root>
`;
